harvester/dmat/src/main/java/dk/dbc/dataio/harvester/dmat/HarvestOperation.java

```diff
@@ -139,30 +191,39 @@ private void assertRecordState(DMatRecord dmatRecord) throws HarvesterException
                 .withTrackingId(String.join(".", "dmat", config.getLogId(),
                         String.valueOf(dmatRecord.getId())))
                 .withSubmitterNumber(JobSpecificationTemplate.SUBMITTER_NUMBER)
                 .withFormat(config.getContent().getFormat())
-                .withCreationDate(Date.from(creationDate.atStartOfDay(timezone).toInstant()));
+                .withCreationDate(Date.from(creationDate.atStartOfDay(timezone).plusHours(12).toInstant()));
+        metaData.setdmatRecord(dmatRecord);
+        return metaData;
+
+        // Note: Using creationDate.atStartOfDay(...) will actually, for us in a timezone with UTC -1 or -2
+        //       hours, put the creationdate stamp at the day before.  This is kind of weird, but is
+        //       the standardized way to handle this in dataio, so we shall not invent a new truth.
+        //
+        // The flowscript handling jobs from the dmat harvester MUST use the 'formattedCreationDate property
+        // provided by the dmat export object.
     }
 
-    // Todo: here we should take, not a DMatRecord, but a finished export object (which includes the
-    //       visible fields of the dmatrecord
-    private AddiRecord createAddiRecord(AddiMetaData addiMetaData, DMatRecord dmatRecord) throws HarvesterException {
-        // The addi record given to the flowscript contains an embedded addi record, which contains, at position 0:
-        // the encapsulated DMatRecord object (with only visible fields), and at position 1: an marcxml record which is
-        // either the existing record to clone, the record pointet to by the DMatRecord for updating or an LU record.
-        // In certain cases, there is no attached marcxml (CREATE+NEW has no clone record)
-        try {
-            return new AddiRecord(
-                    jsonbContext.marshall(addiMetaData).getBytes(StandardCharsets.UTF_8),
-                    new AddiRecord(new byte[0], new byte[0]).getBytes());  // Todo: add embedded addi record
-        } catch (JSONBException e) {
-            LOGGER.error("Unable to marshall ADDI metadata for dmat record {}: {}", dmatRecord.getId(), e.getMessage());
-            throw new HarvesterException("Unable to marshall ADDI metadata for dmat record " + dmatRecord.getId(), e);
-        }
+    private AddiRecord createAddiRecord(RecordServiceConnector recordServiceConnector, AddiMetaDataWithRecord addiMetaData,
+                                        DMatRecord dmatRecord) throws HarvesterException, JsonProcessingException {
+        final ObjectMapper objectMapper = getDMatObjectMapper();
+
+        // Write the DMatRecord out with only those fields visible for export operations
+        String metaData = objectMapper
+                .writerWithView(RecordView.Export.class).writeValueAsString(addiMetaData);
+
+        // Fetch record to use for cloning or to update
+        String content = RecordFetcher.getRecordFor(recordServiceConnector, dmatRecord);
```

Comment:
Kan der ikke være brug for to records?
Uden at vide nok om det: Men skal der ikke være plads til at der er lavet en ny record, som skal merges med en anden?
Dvs det skal være en recordcollection.

Reply:
Der er en særlig updatecode til merge (som bliver til en update)